File: gpt4-o/GPTToDoNotes/app/src/main/java/com/sanad/gpt4o/gpttodonotes/NoteAdapter.java
package com.sanad.gpt4o.gpttodonotes;

import android.content.Intent;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;
import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import java.io.File;
import java.util.List;

public class NoteAdapter extends RecyclerView.Adapter<NoteAdapter.NoteViewHolder> {

    private List<File> noteFiles;

    public NoteAdapter(List<File> noteFiles) {
        this.noteFiles = noteFiles;
    }

    @NonNull
    @Override
    public NoteViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(android.R.layout.simple_list_item_1, parent, false);
        return new NoteViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull NoteViewHolder holder, int position) {
        File noteFile = noteFiles.get(position);
        holder.textView.setText(noteFile.getName());

        holder.itemView.setOnClickListener(v -> {
            Intent intent = new Intent(v.getContext(), NoteDetailActivity.class);
            intent.putExtra("notePath", noteFile.getAbsolutePath());
            v.getContext().startActivity(intent);
        });
    }

    @Override
    public int getItemCount() {
        return noteFiles.size();
    }

    static class NoteViewHolder extends RecyclerView.ViewHolder {
        TextView textView;

        public NoteViewHolder(@NonNull View itemView) {
            super(itemView);
            textView = itemView.findViewById(android.R.id.text1);
        }
    }
}

File: gemini_pro_2.5/ToDoNotes/app/src/main/java/com/sanad/gemini_2_dot_5_pro_preview/todonotes/NoteAdapter.kt
package com.sanad.gemini_2_dot_5_pro_preview.todonotes

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.DiffUtil
import androidx.recyclerview.widget.ListAdapter
import androidx.recyclerview.widget.RecyclerView
import com.sanad.gemini_2_dot_5_pro_preview.todonotes.databinding.ListItemNoteBinding // Import ViewBinding
import java.text.SimpleDateFormat
import java.util.*

class NoteAdapter(private val onClickListener: (Note) -> Unit) :
    ListAdapter<Note, NoteAdapter.NoteViewHolder>(NoteDiffCallback()) {

    private val dateFormat = SimpleDateFormat("MMM d, yyyy", Locale.getDefault())

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): NoteViewHolder {
        val binding = ListItemNoteBinding.inflate(LayoutInflater.from(parent.context), parent, false)
        return NoteViewHolder(binding)
    }

    override fun onBindViewHolder(holder: NoteViewHolder, position: Int) {
        val note = getItem(position)
        holder.bind(note, dateFormat)
        holder.itemView.setOnClickListener { onClickListener(note) }
    }

    class NoteViewHolder(private val binding: ListItemNoteBinding) : RecyclerView.ViewHolder(binding.root) {
        fun bind(note: Note, dateFormat: SimpleDateFormat) {
            binding.textViewNoteTitle.text = note.title
            binding.textViewNoteDate.text = dateFormat.format(Date(note.lastModified))
        }
    }

    class NoteDiffCallback : DiffUtil.ItemCallback<Note>() {
        override fun areItemsTheSame(oldItem: Note, newItem: Note): Boolean {
            return oldItem.id == newItem.id
        }

        override fun areContentsTheSame(oldItem: Note, newItem: Note): Boolean {
            // Compare relevant fields for content changes
            return oldItem.title == newItem.title &&
                    oldItem.lastModified == newItem.lastModified // Simple check
        }
    }
}